File: common/src/main/java/net/impleri/mobskills/integrations/kubejs/MobSkillsPlugin.java
package net.impleri.mobskills.integrations.kubejs;

import dev.latvian.mods.kubejs.KubeJSPlugin;
import net.impleri.mobskills.integrations.kubejs.events.EventsBinding;
import net.impleri.mobskills.integrations.kubejs.events.RestrictionsRegistrationEventJS;
import net.minecraft.server.MinecraftServer;

public class MobSkillsPlugin extends KubeJSPlugin {
    @Override
    public void registerEvents() {
        EventsBinding.GROUP.register();
    }

    public static void onStartup(MinecraftServer server) {
        EventsBinding.RESTRICTIONS.post(new RestrictionsRegistrationEventJS(server));
    }
}

File: common/src/main/kotlin/net/impleri/mobskills/integrations/kubejs/MobSkillsPlugin.kt
package net.impleri.mobskills.integrations.kubejs

import dev.architectury.event.events.common.LifecycleEvent
import dev.latvian.mods.kubejs.KubeJSPlugin
import net.impleri.mobskills.integrations.kubejs.events.EventsBinding
import net.impleri.mobskills.integrations.kubejs.events.RestrictionsRegistrationEventJS
import net.minecraft.server.MinecraftServer

class MobSkillsPlugin : KubeJSPlugin() {
  init {
    LifecycleEvent.SERVER_STARTING.register(
      LifecycleEvent.ServerState { onStartup(it) },
    )
  }

  override fun registerEvents() {
    EventsBinding.GROUP.register()
  }

  companion object {
    fun onStartup(server: MinecraftServer) {
      EventsBinding.RESTRICTIONS.post(
        RestrictionsRegistrationEventJS(server),
      )
    }
  }
}
